FieldOpt/Optimization/optimizers/CMA_ES.cpp

```diff
@@ -0,0 +1,334 @@
+/******************************************************************************
+    Created by Brage on 18/06/19.
+    Copyright (C) 2019 Brage Strand Kristoffersen <brage_sk@hotmail.com>
+
+    This file is part of the FieldOpt project.
+
+    FieldOpt is free software: you can redistribute it and/or modify
+    it under the terms of the GNU General Public License as published by
+    the Free Software Foundation, either version 3 of the License, or
+    (at your option) any later version.
+
+    FieldOpt is distributed in the hope that it will be useful,
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+    GNU General Public License for more details.
+
+    You should have received a copy of the GNU General Public License
+    along with FieldOpt.  If not, see <http://www.gnu.org/licenses/>.
+******************************************************************************/
+
+#include "CMA_ES.h"
+#include "Utilities/math.hpp"
+#include "Utilities/random.hpp"
+#include "Utilities/stringhelpers.hpp"
+#include "Settings/optimizer.h"
+#include <math.h>
+#include <random>
+
+namespace Optimization {
+namespace Optimizers {
+CMA_ES::CMA_ES(Settings::Optimizer *settings,
+               Case *base_case,
+               Model::Properties::VariablePropertyContainer *variables,
+               Reservoir::Grid::Grid *grid,
+               Logger *logger,
+               CaseHandler *case_handler,
+               Constraints::ConstraintHandler *constraint_handler
+) : Optimizer(settings, base_case, variables, grid, logger, case_handler, constraint_handler) {
+    settings_ = settings;
+    // User defined parameters (need to be edited)
+    n_vars_ = variables->ContinousVariableSize();
+    std::random_device rd{};
```

Comment:
You should use the facilities in `Utilities/random.hpp` for generating random numbers. If the distribution you need isn't there, you can easily add it.